lib/upgrade/upgrade_test.go

```diff
@@ -179,9 +178,22 @@ func TestCompatibilityJson(t *testing.T) {
 	if compInfo.Runtime != strings.TrimSpace(compInfo.Runtime) {
 		t.Errorf("%s: %q contains spaces", compatibilityJson, compInfo.Runtime)
 	}
-	i := version.Compare(compInfo.Runtime, runtime.Version())
-	if i > 0 {
-		t.Errorf("Got version %s, want %s in %s", compInfo.Runtime, runtime.Version(), compatibilityJson)
+	// If we're running inside a GitHub action, we need to compare the go
+	// version in compatibility.json with both the current runtime version, and
+	// the previous one, as build-syncthing.yaml runs our unit tests with
+	// both versions.
```

Comment:
So this brings to light the fact that this check is going to be a pain in the ass for anyone wanting to use a different Go version for whatever reason and not planning to distribute Syncthing or care about the compat file. We should probably have this test run only with a specific `release` tag or something, to make sure _we_ run it but others don't have to.

Reply:
True. We could add an env var to either opt-in or opt-out. Perhaps opt-out with `STSKIP_COMPATIBILTY_CHECK`?

A dev could always tweak `compatibility.json` to match their env, or even delete the file. Of course, they would need to be careful not to commit their tweak.

Reply:
Good idea, I'll add that.

A dev could also tweak `compatibility.json` to match their env, or even delete the file. Of course, they would need to be careful not to commit their tweak.